src/components/landingPage/index.js

```diff
@@ -0,0 +1,270 @@
+    const oldNumber = state[viewMoreArticlesButton];
+    const initialLimit = state.initialLimits[viewMoreArticlesButton];
+    const newNumber = oldNumber >= articlesNumber ? initialLimit : oldNumber + number;
+    this.setState((prevState) => ({
+      ...prevState,
+      [viewMoreArticlesButton]: newNumber,
+    }));
+  };
+
+  render() {
+    const {
+      articles,
+      popularArticles,
+      limitArticlesNumber,
+      limitPopularArticlesNumber,
+      loading,
+    } = this.state;
+    const mostTwoRecentArticles = articles.slice(0, 2);
+    const secondAndThirdArticles = articles.slice(2, 4);
+    const remainingArticles = articles.slice(4);
+    const displayFirstAndSecond = mostTwoRecentArticles.map((article) => (
+      <a
+        href={`${actionTypes.BACKEND_URL}/api/${actionTypes.VERSION}/articles/${article
+          && article.slug}`}
+      >
+        <ArticleCard
+          key={article && article.slug}
+          coverImage={article && article.images && article.images[0]}
+          description={article && article.description}
+          title={article && article.title}
+          views={article && article.views}
+          readTime={article && article.readtime}
+          userIcon={article && article.userImage}
+          username={article && article.username}
+          createdTime={article && article.timeCreated}
+        />
+      </a>
+    ));
+
+    const displayThirdAndFourth = secondAndThirdArticles.map((article) => (
+      <a
+        href={`${actionTypes.BACKEND_URL}/api/${actionTypes.VERSION}/articles/${article
+          && article.slug}`}
+      >
+        <ArticleCard
+          key={article && article.slug}
+          coverImage={article && article.images && article.images[0]}
+          description={article && article.description}
+          title={article && article.title}
+          views={article && article.views}
+          readTime={article && article.readtime}
+          userIcon={article && article.userImage}
+          username={article && article.username}
+          createdTime={article && article.timeCreated}
+        />
+      </a>
+    ));
+    const otherArticles = remainingArticles.map((article, index) => {
+      if (index >= limitArticlesNumber) return false;
+      return (
+        <a
+          href={`${actionTypes.BACKEND_URL}/api/${actionTypes.VERSION}/articles/${article
+            && article.slug}`}
+        >
+          <ArticleCard
+            key={article && article.slug}
+            description={article && article.description}
+            coverImage={article && article.images && article.images[0]}
+            title={article && article.title}
+            views={article && article.views}
+            readTime={article && article.readtime}
+            userIcon={article && article.userImage}
+            username={article && article.username}
+            createdTime={article && article.timeCreated}
+          />
+        </a>
+      );
+    });
```

Comment:
There is a lot of repetition here. You should create a component called ```articlesList``` which maps the list given in its props to the article card

Reply:
I worked on the feedback